[BUG] `allclose` asserts with "'array' object has no attribute 'typecode'" when arrays aren't all close

I have the following code which is throwing 

```
>       assert mx.allclose(time_rope_out, mlx_rope_out)
E       assert <[AttributeError("'array' object has no attribute 'typecode'") raised in repr()] array object at 0x1056c0300>
E        +  where <[AttributeError("'array' object has no attribute 'typecode'") raised in repr()] array object at 0x1056c0300> = <nanobind.nb_func object at 0x105787670>(<[ValueError('[convert] Only length-1 arrays can be converted to Python scalars.') raised in repr()] array object at 0x105cdc5c0>, <[ValueError('[convert] Only length-1 arrays can be converted to Python scalars.') raised in repr()] array object at 0x1056f7d80>)
E        +    where <nanobind.nb_func object at 0x105787670> = mx.allclose
```

in testing. I //think// I just have my atol/rtol set wrong (or a bug :D), but that's a weird error message. If I twiddle atol or just write allclose manually it seems ok.

Code to repro:

```python
from mlx import nn
import mlx.core as mx
from einops.array_api import rearrange, repeat


def rotate_half(x):
    x = rearrange(x, '... (d r) -> ... d r', r = 2)
    x1 = x[..., 0]
    x2 = x[..., 1]
    x = mx.stack((-x2, x1), axis = -1)
    return rearrange(x, '... d r -> ... (d r)')

def apply_rotary_emb(freqs, tensor):
    dtype = tensor.dtype
    if freqs.ndim == 2:
        freqs = freqs[None, None, ...]
    if freqs.ndim == 3:
        freqs = freqs[:, None, ...]
    t_transformed = (tensor * freqs.cos()) + (rotate_half(tensor) * freqs.sin())
    return t_transformed.astype(dtype)

class TimeRotaryEmbedding(nn.Module):
    def __init__(
        self,
        dim,
        max_t = 10,
    ):
        super().__init__()
        self.dim = dim
        self.max_t = max_t

    def __call__(self, tensor, time):
        time = time - time.min(axis=-1, keepdims=True)
        freqs = self.freqs(time)
        return apply_rotary_emb(freqs, tensor)

    def freqs(
        self,
        time
    ):
        freqs = 1. / (self.max_t ** (mx.arange(0, self.dim, 2)[:(self.dim // 2)].astype(mx.float32) / self.dim))
        freqs = mx.einsum('b s, f -> b s f', time.astype(freqs.dtype), freqs)
        freqs = repeat(freqs, '... n -> ... (n r)', r = 2)
        return freqs



def test_time_rope_same_as_normal_rope():
    base = 10000
    time_rope = TimeRotaryEmbedding(dim = 6, max_t=base)
    mlx_rope = nn.RoPE(dims = 6, traditional=True, base=base)

    x = mx.random.normal((1, 4, 12, 6)) # (batch_size, num_heads, seq_len, dim)
    time = mx.arange(0, 12)[None, :]
    time_rope_out = time_rope(x, time=time)
    mlx_rope_out = mlx_rope(x)
    assert mx.allclose(time_rope_out, mlx_rope_out)
```

--- a/python/mlx/_reprlib_fix.py
+++ b/python/mlx/_reprlib_fix.py
@@ -3,18 +3,14 @@
 import array
 import reprlib
 
-
-class FixedRepr(reprlib.Repr):
-    """Only route python array instances to repr_array."""
-
-    def repr_array(self, x, maxlevel):
-        if isinstance(x, array.array):
-            return super().repr_array(x, maxlevel)
-        else:
-            return self.repr_instance(x, maxlevel)
+_old_repr_array = reprlib.Repr.repr_array
 
 
-# We need to monkey-patch reprlib so that we can use the debugger without
-# renaming the array to something else
-fixed_repr = FixedRepr()
-reprlib.repr = fixed_repr.repr
+def repr_array(self, x, maxlevel):
+    if isinstance(x, array.array):
+        return _old_repr_array(self, x, maxlevel)
+    else:
+        return self.repr_instance(x, maxlevel)
+
+
+reprlib.Repr.repr_array = repr_array

